frontend/src/APIClients/ResidentAPIClient.ts

```diff
@@ -0,0 +1,29 @@
+import AUTHENTICATED_USER_KEY from "../constants/AuthConstants";
+import { getLocalStorageObjProperty } from "../utils/LocalStorageUtils";
+import baseAPIClient from "./BaseAPIClient";
+
+import { Resident } from "../types/ResidentTypes";
+
+const getResidents = async (): Promise<any> => {
+  let residentData = [];
+  try {
+    const bearerToken = `Bearer ${getLocalStorageObjProperty(
+      AUTHENTICATED_USER_KEY,
+      "accessToken",
+    )}`;
+    const { data } = await baseAPIClient.get("/residents/", {
+      headers: { Authorization: bearerToken },
+    });
+    if (data) {
+      residentData = data;
+    }
```

Comment:
Doesn't this route return an empty array by default? If not, we should change it to so it's not on the responsibility of the frontend